src/pages/ProfilesExplorerView/components/SceneLabelValuesTimeseries/SceneLabelValuesTimeseries.tsx

```diff
@@ -145,6 +173,37 @@ export class SceneLabelValuesTimeseries extends SceneObjectBase<SceneLabelValues
     }
   }
 
+  private subscribeToExemplarToggleChanges() {
+    const { exemplarToggleAction } = this.state;
+    if (!exemplarToggleAction) {
+      return null;
+    }
+
+    return exemplarToggleAction.subscribeToState((newState, prevState) => {
+      if (newState.showExemplars !== prevState.showExemplars) {
+        this.handleExemplarToggleChange(newState.showExemplars ?? false);
+      }
+    });
+  }
+
+  private handleExemplarToggleChange(includeExemplars: boolean) {
+    const { body, item, displayAllValues, annotations } = this.state;
+
+    const { queries } = buildTimeSeriesQueryRunner(
+      item.queryRunnerParams,
+      displayAllValues ? undefined : LabelsDataSource.MAX_TIMESERIES_LABEL_VALUES,
+      annotations,
+      includeExemplars
+    ).state;
+
+    const queryRunner = body.state.$data?.state.$data as SceneQueryRunner;
+
+    if (queryRunner) {
+      queryRunner.setState({ queries });
+      queryRunner.runQueries();
```

Comment:
If I understood the code properly, we are fetching data everytime we toggle Exemplars. I think isn't necssary when disabling;  we can just hide the exemplars without refetching the data.